patches/minecraft/net/minecraft/enchantment/EnchantmentHelper.java.patch

```diff
@@ -1,6 +1,25 @@
 --- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
 +++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
-@@ -302,7 +302,7 @@
+@@ -143,12 +143,14 @@
+         return field_77520_b.field_77497_a;
+     }
+ 
++    /**
++     * Does nothing
++     * Use {@link EnchantmentHelper#getModifierForCreature(ItemStack, Entity)} instead
++     */
++    @Deprecated
+     public static float func_152377_a(ItemStack p_152377_0_, EnumCreatureAttribute p_152377_1_)
+     {
+-        field_77521_c.field_77495_a = 0.0F;
+-        field_77521_c.field_77494_b = p_152377_1_;
+-        func_77518_a(field_77521_c, p_152377_0_);
+-        return field_77521_c.field_77495_a;
++        return 0.0F;
```

Comment:
Unobfuscated old code:
``` java
        ENCHANTMENT_MODIFIER_LIVING.livingModifier = 0.0F;
        ENCHANTMENT_MODIFIER_LIVING.entityLiving = creatureAttribute;
        applyEnchantmentModifier(ENCHANTMENT_MODIFIER_LIVING, stack);
        return ENCHANTMENT_MODIFIER_LIVING.livingModifier;
```

The old code is not working anymore since the ENCHANTMENT_MODIFIER_LIVING now requires an Entity instead of an EnumCreatureAttribute to calculate it's value. Since this is not available in this context I decided to just return 0 in this deprecated method.

This might actually cause minor problems with mods that do use this method directly instead of indirectly via EntityPlayer or EntityMob as those mods would not find out the actual value for some enchantments.

However, the only other (non breaking) solution I can think of, is to create second ENCHANTMENT_MODIFIER_(CREATURE_OLD) (or integrate it in the current one) which can make use of a EntityCreatureAttribute. But since this would ignore any values added to the enchantments in newer ways this would be a somewhat ugly solution.